docker_cli/commands.py

```diff
@@ -66,6 +67,61 @@ def get_archive(
     return Image.build(tag=img_tag, dockerfile_string=dockerfile, no_cache=True)
 
 
+def insert(tag: str, base: str, directory: str | os.PathLike[str]):
+    """
+    Builds a Docker image with the contents of the given directory copied onto it.
+
+    The directory path on the image has the same name as the topmost directory
+    of the given path. e.g. giving path "/tmp/dir/subdir" will result in the contents of
+    this path being saved in "/subdir" on the generated image.
+
+    This Dockerfile also places the working directory of the image inside of the copied
+    directory.
+
+    Parameters
+    ----------
+    tag : str
+        The image tag.
+    base : str
+        The base image tag.
+    directory : path-like
+        The directory to be copied.
+
+    Returns
+    -------
+    Image
+        The generated image.
+    """
+
+    prefix = universal_tag_prefix()
+    img_tag = tag if tag.startswith(prefix) else f"{prefix}-{tag}"
+
+    dir_str = str(directory)
+
+    # The absolute path of the given directory will be the build context.
+    # This is necessary because otherwise docker may be unable to find the directory if
+    # the build context is at the current working directory.
+    path_absolute = os.path.abspath(str(dir_str))
```

Comment:
`dir_str` is already a string

```suggestion
    path_absolute = os.path.abspath(dir_str)
```